custom-routes-vis/src/main/webapp/script.js

```diff
@@ -12,9 +12,105 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/** Creates the world map. */
+// Array holding origin and destination markers.
+let markers = []
+
+/** Creates the world map and markers. */
 function createWorldMap() {
   map = new google.maps.Map(
     document.getElementById('map'),
     {center: {lat: 38.46049, lng: -5.428423}, zoom: 3});
+  
+  // Create origin and destination markers, but hides them until user selects
+  // coordinates.
+  createMarker('origin-coordinates', 'A', 'Origin',  {lat: 0.0, lng: 0.0});
+  createMarker('destination-coordinates', 'B', 'Destination',
+      {lat: 0.0, lng: 0.0});
+}
+
+/**
+ * Creates hidden marker that updates coordiantes on page when moved.
+ * @param {string} containerId id of container to update coordinates in.
+ * @param {string} label Label to display on container.
+ * @param {Object} latLng Default coordinates of where to place marker.
+ */
+function createMarker (containerId, label, title, latLng) {
+  let marker = new google.maps.Marker({
+    draggable: true,
+    label: label,
+    map: map,
+    position: latLng,
+    title: title,
+    visible: false,
+  });
+  // Add marker to markers array to access in other functions.
+  markers.push(marker);
+
+  marker.addListener('dragend', function(event){
+    updateCoordinates(event.latLng.lat(), event.latLng.lng(), containerId);
+  });
 }
+
+/**
+ * Updates the displayed coordinates on the page.
+ * @param {string} containerId id of container to update coordinates in.
+ */
+function updateCoordinates(lat, lng, containerId) {
+  document.getElementById(containerId).innerHTML = 
+      "Latitude: " + lat + "<br>Longitude: " + lng + "<br>";
+}
+
+/**
+ * Hides 'place marker' button and waits for user to select coordinate for
+ * marker. Updates marker to be in correct location, and show 'delete marker'
+ * button.
+ * @param {int} markerIndex Index of corresponding marker in markers array.
```

Comment:
Instead of an index? How about using an enum. This would make things a lot cleaner and also extendible.